### ISSUE TITLE
Synchronous `pre( 'save' )` is not called in the context of a doc.

### ISSUE THREAD

**Issue by AshHeskes** [neumino/thinky#354]
When adding a sync pre save hook when the hook is called it is not being called in the context of the doc it belongs to. Instead it's being called in the context of the calling function that created the model.

As soon as you add a `next` argument to make it async it is called in the correct context of doc.

```js
var model = thinky.createModel({});

model.pre( 'save', function(){
	console.log(this);	
})

model.pre( 'save', function(next){
	console.log(this);	
})
```

**Comment by JedWatson** [comment]
Just in case you're interested, we (by which I really mean @creynders) developed a solid package called [grappling-hook](https://github.com/keystonejs/grappling-hook) for KeystoneJS that abstracts this sort of thing.

It might be worth looking into for thinky, since it's comprehensively documented and tested, and should be fairly easy to drop in. Supports sync / async and Promise-based hooks.

**Comment by neumino** [comment]
Thanks @JedWatson, I'll take a look at that when I get a few minutes.

**Comment by AshHeskes** [comment]
@JedWatson Looks really good!

**Comment by creynders** [comment]
@neumino if you have any questions, let me know!!

**Comment by neumino** [comment]
Fixed in 2.1.9
